### ISSUE TITLE
Idea: export startBatch/endBatch

### ISSUE THREAD

**Issue by mivion** [mobxjs/mobx#1375]
Hi!

May be this is bad idea, but what if mobx exports startBatch/endBatch?
This will resolve many problems with async/await actions.

For example the code:
```jsx
@observable count = 0;

@action someAction = async () => {
      this.count ++; // render/reaction will be called!
      await makeFetch ();
      this.count ++; // render/reaction will be called!
      await makeFetch ();
      this.count ++; // render/reaction will be called!
}

render () {
   return <div>{this.count}</div>
}
```
So I need that render/reaction will be called only after *action* finished. Even asyncActions from mobx-utils doesn't help in this situation, because between next ()/yield calls render/reaction are called.
But if mobx exports startBatch/endBatch, *action* works as expected:

```jsx
@observable count = 0;

@action someAction = async () => {
      startBatch ();
      this.count ++; // render/reaction will not be called!
      await makeFetch ();
      this.count ++; // render/reaction will not be called!
      await makeFetch ();
      this.count ++; // render/reaction will not be called!
      endBatch ();
      // render/reaction will be called!
}

render () {
   return <div>{this.count}</div>
}
```

I agree that this is hack, but for now there no way to start/stop transactions.

For example it would be nice to send callback to transaction/action when its done:
```jsx
transaction (async (done) => {
   this.count ++;
   await someFetch ();
   this.count ++;
   await someFetch ();
   done (); // we are done
});
```

Or just add support async/await for transaction/action api.

May be this helps.

**Comment by mweststrate** [comment]
The problem of what you are proposing is that an unfinished batch doesn't prevent to distributing your changes; it prevents distributing *any* changes. So until both fetches complete, your app will be literally frozen. 

Instead, just don't change the state until you are willing to commit to your state changes. If you don't want your intermediate count values to be distribute, why bother updating the state in the first place.

E.g. you could simply do: 

```javascript
@action someAction = async () => {
      let toAdd = 1 
      await makeFetch ();
      toAdd ++;
      await makeFetch ();
      this.count += toAdd
}
```

I understand this example is contrived, but note that this pattern can easily be generalized by storing "stuff to do" thunks

```javascript
@action someAction = async () => {
      const tasks = []
      tasks.push(() => { this.count++ })
      await makeFetch ();
      tasks.push(() => { this.count++ })
      await makeFetch ();
      tasks.push(() => { this.count++ })
      tasks.forEach(f => f())
}
```

This way you also postpone all state changes until the end of your async process, but without breaking the MobX model that state is always synchronously distributed and can never be observed to be stale.
